### PR TITLE
Split command line using quote respecting shlex.split instead of ' '.split

### PR DESCRIPTION
**Pull request by tclose**
## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Summary
Fixes an issue when there are quotes in argument strings to enclose spaces and other special characters.

## Checklist

- [x] I have added tests to cover my changes (if necessary)
- [x] I have updated documentation (if necessary)

### CODE REVIEW

File: pydra/engine/task.py [hunk 1]
```diff
@@ -519,10 +518,19 @@ def _run_task(self):
...
                 if self.output_["stderr"]:
-                    raise RuntimeError(self.output_["stderr"])
-                else:
-                    raise RuntimeError(self.output_["stdout"])
+                    msg += "\n\nstderr:\n" + self.output_["stderr"]
+                if self.output_["stdout"]:
+                    msg += "\n\nstderr:\n" + self.output_["stdout"]
+                raise RuntimeError(msg)
+
+    def _join_cmd_args(self, args):
```
Comment (tclose) [inline pydra/engine/task.py:528]: Not 100% sure about this, feels a bit hacky just skipping the executable but it should work. Any suggestions on a more elegant solution?

**Comment by tclose** [comment]
Hi @djarecka, could you have a look at this when you get the chance? Basically uses shlex.split instead of str.split(' ') when creating command args. This allows you to put quotes around cmd args and handle special cases like spaces in file names.

 The tests that are failing seem to be some issue with the GH action runner rather than my code if you look at them.

**Comment by tclose** [comment]
Ah, so it seems I have some Windows errors, which is annoying as I don't have a Windows test box atm. Seems as though the Windows path backslashes are getting eaten by the Python string. Will try to work out why that might be.

NB: the code-cov patch is down because I have added a switch to the conftest to allow you to enable the raising of exceptions while debugging unittests

**Comment by djarecka** [comment]
Hmm, not sure why windows raises the error just after reading it (also don't have windows)...
But I'm also not sure if I understand the changes in pytest - seems like no test from `test_shelltask.py` was run...

**Comment by effigies** [comment]
I think we're getting bitten by https://github.com/python/cpython/issues/44990. Might be worth looking for a windows-compatible shlex.

**Comment by effigies** [comment]
Actually, what about using: 

```Python
import sys
import shlex

def shell_split(val):
    return shlex.split(var, posix="win" not in sys.platform)
```

Via https://github.com/jrnl-org/jrnl/issues/348

**Comment by tclose** [comment]
Nice find @effigies! Although I had to change their clause because it was matching "dar**win**" :) 

All checks are passing now except the code-cov due to the debugging clause I inserted into conftest. Happy to take this out if you are not a fan.

**Comment by tclose** [comment]
Ok, I believe this is good to go from my end. Just 1.6% down on the coverage patch

**Comment by djarecka** [comment]
Thanks @tclose !

I think we can keep the the debugging clause.

However I do worry that you are having issue with running `test_shelltask.py` on your laptop. Could you create an issue and show the error?

**Comment by tclose** [comment]
Thanks @djarecka, actually after I reinstalled pydra[dev] and reconfigured my VSCode pytest all the tests showed up, so it must have just been something that needed refreshing.

**Pull request merged**